ansible_collections/arista/avd/roles/eos_designs/python_modules/overlay/utils.py

```diff
@@ -257,3 +257,106 @@ def _append_peer(self, peers_dict: dict, peer_name: str, peer_facts: dict) -> No
+        sites = get(self._wan_region, "sites", [])
+        return get_item(sites, "name", node_defined_site, required=True)
+
+    @cached_property
+    def _wan_region(self) -> dict | None:
+        """
+        WAN region for Pathfinder
+        """
+        if not self.shared_utils.cv_pathfinder_role:
+            return None
+
+        node_defined_region = get(self.shared_utils.switch_data_combined, "cv_pathfinder_region", required=True)
+        regions = get(self._hostvars, "cv_pathfinder_regions", required=True)
+
+        return get_item(regions, "name", node_defined_region, required=True)
+
+    @cached_property
+    def _wan_zone(self) -> dict | None:
+        """
+        WAN zone for Pathfinder
+
+        Currently, only default zone DEFAULT-ZONE with ID 1 is supported.
+        """
+        if not self.shared_utils.cv_pathfinder_role:
+            return None
+
+        # Injecting zone DEFAULT-ZONE with id 1.
+        return {"name": "DEFAULT-ZONE", "id": 1}
+
+    @cached_property
+    def _wan_route_servers(self) -> dict:
+        """
+        Return a list of wan RR based on the the wan_mode type
+
+        It the RR is part of the inventory, the peer_facts are read.
+        If any key is specified in the variables, it overwrites whatever is in the peer_facts.
+
+        If no peer_fact is found the variables are required in the inventory.
+        """
+        # TODO - need to factor this with other function once we fix
+        # https://github.com/aristanetworks/ansible-avd/issues/3392
+        if not self.shared_utils.wan_mode:
+            return {}
+
+        wan_route_servers = {}
+
+        wan_route_servers_list = get(self._hostvars, "wan_route_servers", default=[])
+
+        for wan_rr_dict in natural_sort(wan_route_servers_list, sort_key="hostname"):
+            # These remote gw can be outside of the inventory
+            wan_rr = wan_rr_dict["hostname"]
+
+            if wan_rr == self.shared_utils.hostname:
+                # Don't add yourself
+                continue
+
+            peer_facts = self.shared_utils.get_peer_facts(wan_rr, required=False)
+
+            wan_rr_result_dict = {}
+
+            # TODO need to test this
+            if peer_facts is not None:
+                # Found a matching server in inventory
+                bgp_as = peer_facts.get("bgp_as")
+
+                wan_rr_result_dict = {
+                    "bgp_as": str(bgp_as) if bgp_as is not None else None,
+                    "router_id": peer_facts.get("router_id"),
+                    "carriers": peer_facts.get("carriers"),
+                }
+
+            # If overlay_routing_protocol is iBGP then we can use our own AS
+            if self.shared_utils.overlay_routing_protocol == "ibgp":
+                bgp_as = self.shared_utils.bgp_as
+
+            # Retrieve the values from the dictionary, making them required if the peer_facts were not found
+            # TODO needs to be improved
```

Comment:
You could get the vars with default to the peer facts under the 331 if.
And as an else you get the vars with required=True.

Reply:
so I implemented this but then I need to add extra line to check - that the value returned from `get` is not None in case of peer facts so .. not necessarily better